File: binaryTreeOrderTraversal.js
var levelOrder = function(root) {
    let result = []
    if(!root){
        return result
    }
    let q = [root]
    while(q.length > 0){
        let levelVals = []
        let nextLevel = []
        for(let node of q){
            levelVals.push(node.val)
            if(node.left){
                nextLevel.push(node.left)
            }
            if(node.right){
                nextLevel.push(node.right)
            }
        }
        result.push(levelVals)
        q = nextLevel
    }
    return result
};

File: binaryTreeOrderTraversal.py
def levelOrder(self, root: TreeNode) -> List[List[int]]:
    result = []
    if root is None:
        return result
    q = [root]
    while len(q) > 0:
        levelVals = []
        level = []
        for node in q:
            levelVals.append(node.val)
            if node.left:
                level.append(node.left)
            if node.right:
                level.append(node.right)
        result.append(levelVals)
        q = level
    return result
